app/ide-desktop/lib/content/src/index.ts

```diff
@@ -180,60 +180,60 @@ class Main implements AppRunner {
-                const ide = document.getElementById('root')
-                if (auth) {
-                    auth.style.display = 'none'
-                }
-                if (ide) {
-                    ide.hidden = false
+        const unrecognized = contentConfig.OPTIONS.loadAll([app.urlParams()])
+        if (unrecognized.length > 0) {
+            app.showConfigOptions(contentConfig.OPTIONS, unrecognized)
+        } else {
+            const isUsingAuthentication = contentConfig.OPTIONS.options.authentication.value
+            const isUsingNewDashboard =
+                contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value
+            const isOpeningMainEntryPoint =
+                contentConfig.OPTIONS.groups.startup.options.entry.value ===
+                contentConfig.OPTIONS.groups.startup.options.entry.default
+            if ((isUsingAuthentication || isUsingNewDashboard) && isOpeningMainEntryPoint) {
+                const hideAuth = () => {
+                    const auth = document.getElementById('dashboard')
+                    const ide = document.getElementById('root')
+                    if (auth) {
+                        auth.style.display = 'none'
+                    }
+                    if (ide) {
+                        ide.hidden = false
+                    }
                 }
-            }
-            /** This package is an Electron desktop app (i.e., not in the Cloud), so
-             * we're running on the desktop. */
-            /** TODO [NP]: https://github.com/enso-org/cloud-v2/issues/345
-             * `content` and `dashboard` packages **MUST BE MERGED INTO ONE**. The IDE
-             * should only have one entry point. Right now, we have two. One for the cloud
-             * and one for the desktop. */
-            const currentPlatform = contentConfig.OPTIONS.groups.startup.options.platform.value
-            let platform = authentication.Platform.desktop
-            if (currentPlatform === 'web') {
-                platform = authentication.Platform.cloud
-            }
-            /** FIXME [PB]: https://github.com/enso-org/cloud-v2/issues/366
-             * React hooks rerender themselves multiple times. It is resulting in multiple
-             * Enso main scene being initialized. As a temporary workaround we check whether
-             * appInstance was already ran. Target solution should move running appInstance
-             * where it will be called only once. */
-            let appInstanceRan = false
-            const onAuthenticated = () => {
-                if (!contentConfig.OPTIONS.groups.featurePreview.options.newDashboard.value) {
-                    hideAuth()
-                    if (!appInstanceRan) {
-                        appInstanceRan = true
-                        void this.runApp(inputConfig)
+                /** This package is an Electron desktop app (i.e., not in the Cloud), so
+                 * we're running on the desktop. */
```

Comment:
As I've written previously (I know this is an old comment), the comment is not true. Even if running in electron, you can connect to localhost via web browser, so the platform will be "web" then. So this is not true that this is desktop platform.